### PR TITLE
Treat symlinks as normal files in spec file

### PR DESCRIPTION
**Pull request by dpennell**
Symlinks are now treated like normal files in the spec.  All of the special handling in the scriptlets has been removed.  Symlinks are now handled properly by upgrades.

### CODE REVIEW

**Comment by muuki88** [comment]
Thanks 😊 I'll take a closer when I come home tomorrow

**Comment by muuki88** [comment]
cc @fsat can you take a look? Would be nice 😊

**Comment by muuki88** [comment]
@dpennell I would first merge #895 and release this. Sorry for the short answer above, I'm not back from vacation and can look into this in more detail.

@fsat implemented a great part of the initial relocation implementation, so it would be nice if he could validate your changes. Maybe he also tried this approach first, but didn't worked out for him.

From my perspective I like the implementation. For two reasons
1. Removing bash code ( always good )
2. Staging directory looks more like the final system installation

**Comment by dpennell** [comment]
I am certainly not an RPM expert.  I based my solution on what appeared to be well informed advice (see comments in #836) to just treat symlinks as regular files (which they are).  I definitely think this needs a serious review.  I've tested on my centos VM and it is being used in our product (currently in QA).

@muuki88 I *think* that this solution would remove the need for #895.

**Comment by muuki88** [comment]
Yes, I hope this will remove the need for #895.

I will test this my self, but I also would like to wait for @fsat review on this.
@huntc Conductr has used some of the relocation features as well, right?

File: src/sbt-test/rpm/symlink-rpm/build.sbt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+linuxPackageSymlinks := {
+  val helloSymlink = LinuxSymlink(((file(defaultLinuxInstallLocation.value) / (packageName in Linux).value / "lib") / "hello.link").toString, "/fake/hello.txt")
+  Seq(helloSymlink)
+}
+
+TaskKey[Unit]("check-spec-file") <<= (target, streams) map { (target, out) =>
+  val spec = IO.read(target / "rpm" / "SPECS" / "rpm-package.spec")
+  out.log.success(spec)
+  assert(spec contains "%attr(0644,root,root) /usr/share/rpm-package/lib/rpm-test.rpm-test-0.1.0.jar", "Wrong installation path\n" + spec)
+  assert(spec contains "/usr/share/rpm-package/lib/hello.link", "Missing or incorrect symbolic link\n" + spec)
```
Comment (muuki88) [inline src/sbt-test/rpm/symlink-rpm/build.sbt]: Wrong links appear in the spec like this?

**Comment by muuki88** [comment]
Closing this for #916 as I can't update this PR.

**Pull request closed without merging**